Description: patch build process (the hard way) to use system hiredis lib
Forwarded: not-needed
Author: Marc Haber <mh+debian-packages@zugschlus.de>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2024-03-24

--- a/builder/MyBuilder.pm
+++ b/builder/MyBuilder.pm
@@ -13,8 +13,8 @@
         generate_ppport_h    => 'src/ppport.h',
         c_source             => 'src',
         xs_files             => { './src/Redis__Fast.xs' => './lib/Redis/Fast.xs', },
-        include_dirs         => ['src', 'deps/hiredis'],
-        extra_linker_flags   => ["deps/hiredis/libhiredis$Config{lib_ext}", "deps/hiredis/libhiredis_ssl$Config{lib_ext}", "-lssl", "-lcrypto"],
+        include_dirs         => ['src', '/usr/include', '/usr/include/hiredis'],
+        extra_linker_flags   => ["-lhiredis", "-lhiredis_ssl", "-lssl", "-lcrypto"],
 
         test_requires => {
             "Digest::SHA"           => "0",
@@ -41,12 +41,6 @@
     } else {
         $make = $Config{make};
     }
-    if (-e '.git') {
-        unless (-e 'deps/hiredis/Makefile') {
-            $self->do_system('git','submodule','update','--init');
-        }
-    }
-    $self->do_system($make, '-C', 'deps/hiredis', 'static', 'USE_SSL=1');
     return $self;
 }
 
--- a/src/Redis__Fast.xs
+++ b/src/Redis__Fast.xs
@@ -4,8 +4,8 @@
 #include "XSUB.h"
 
 #include "ppport.h"
-#include "hiredis.h"
-#include "hiredis_ssl.h"
+#include <hiredis.h>
+#include <hiredis_ssl.h>
 #include "async.h"
 
 #include <string.h>
